mealie/services/parser_services/ingredient_parser.py

```diff
@@ -30,19 +67,70 @@ def parse_one(self, ingredient_string: str) -> ParsedIngredient:
     def parse(self, ingredients: list[str]) -> list[ParsedIngredient]:
         ...
 
+    def find_food_match(self, food: IngredientFood | CreateIngredientFood) -> IngredientFood | None:
+        if isinstance(food, IngredientFood):
+            return food
+
+        query = PaginationQuery(page=1, per_page=10)
+        response = self.foods.page_all(query, search=food.name)
+        if not response.items:
+            return None
+
+        # further refine match using fuzzy matching
+        choices_by_name = {item.name: item for item in response.items}
+        fuzz_result = process.extractOne(food.name, choices_by_name.keys(), scorer=fuzz.ratio)
+        if fuzz_result is None:
+            return None
+
+        choice_name, score, _ = fuzz_result
+        if score < self.food_fuzzy_match_threshold:
+            return None
+        else:
+            return choices_by_name[choice_name]
```

Comment:
I think this should be done before the fuzz searching? Causing a O(1) find if the item exist with an exact match


```py
        choices_by_name = {item.name: item for item in response.items}
        if choice_name in choices_by_name:
                return choices_by_name [choice_name]
        fuzz_result = process.extractOne(food.name, choices_by_name.keys(), scorer=fuzz.ratio)
```